ChatQnA/chatqna.py

```diff
@@ -173,14 +183,22 @@ def align_generator(self, gen, **kwargs):
     yield "data: [DONE]\n\n"
 
 
-class ChatQnAService:
+class ChatQnAService(Gateway):
     def __init__(self, host="0.0.0.0", port=8000):
         self.host = host
         self.port = port
         ServiceOrchestrator.align_inputs = align_inputs
         ServiceOrchestrator.align_outputs = align_outputs
         ServiceOrchestrator.align_generator = align_generator
-        self.megaservice = ServiceOrchestrator()
+
+        super().__init__(
```

Comment:
The graph definition should run before the gateway is started, otherwise the DAG is not recognized correctly by the gateway and the gateway just will be forked to another process without the graph definition. One change could be e.g.

```
        self.megaservice = ServiceOrchestrator()
        self.add_remote_service()   ## Call here to make sure megaservice know the graph
        super().__init__(
            megaservice=self.megaservice,
            host=self.host,
            port=self.port,
            endpoint="/v1/audioqna",
            input_datatype=AudioChatCompletionRequest,
            output_datatype=ChatCompletionResponse,
        )
```

Reply:
update code, please review again~ Thanks~

Reply:
I've reproduced the subsequent CI issues. I guess we need to set `MEGA_SERVICE_HOST_IP=0.0.0.0` after this modification.